### DISCUSSION TITLE
How to run lammps with hybrid  LJ interaction and DP potential

### DISCUSSION THREAD

**Discussion by Nana-bai** [deepmodeling/deepmd-kit#2395] [Q&A]
When I use the nffc.pd model ( 5 atomic types ) hybrid with lj  potential, the simulation can be carried out. When using the myc.pb model ( 3 atomic types ), the simulation will appear "core dump "and exit,  the total atomic type of the simulation system is 5.
![00967dbce5859c22167f4437d56dd9c](https://user-images.githubusercontent.com/127933368/225249004-0fbb7f7e-5c3a-4bec-8f1f-072376b4c4bc.png)
![image](https://user-images.githubusercontent.com/127933368/225249152-27050d35-545d-412e-9199-d07becc64d9e.png)
LAMMPS (29 Sep 2021 - Update 3)
OMP_NUM_THREADS environment is not set. Defaulting to 1 thread. (src/comm.cpp:98)
  using 1 OpenMP thread(s) per MPI task
Loading plugin: deepmd pair style v2.0 by Han Wang
Loading plugin: compute deeptensor/atom v2.0 by Han Wang
Loading plugin: fix dplr v2.0 by Han Wang
units                 metal
dimension             3
boundary              p  p  p

atom_style            full
bond_style            harmonic
angle_style           harmonic
dihedral_style        harmonic

#read_data             NaCaCl-SiO2.data

read_restart  lj.restart
Reading restart file ...
  restart file = 29 Sep 2021, LAMMPS = 29 Sep 2021
WARNING: Restart file used different # of processors: 32 vs. 1 (src/read_restart.cpp:658)
  restoring atom style full from restart
  orthogonal box = (-2.5152592 -2.5152592 -2.5152592) to (72.515259 72.515259 72.515259)
  1 by 1 by 1 MPI processor grid
  restoring pair style lj/cut/coul/long from restart
  restoring bond style harmonic from restart
  restoring angle style harmonic from restart
  restoring dihedral style harmonic from restart
  10522 atoms
  466 bonds
  842 angles
  1124 dihedrals
  22 impropers
Finding 1-2 1-3 1-4 neighbors ...
  special bond factors lj:    0        0        0       
  special bond factors coul:  0        0        0       
     4 = max # of 1-2 neighbors
     6 = max # of 1-3 neighbors
    16 = max # of 1-4 neighbors
    20 = max # of special neighbors
  special bonds CPU = 0.003 seconds
  read_restart CPU = 0.009 seconds

region                fixed  sphere 34.2761 37.7968 34.2943 3

mass                  1      23.000000
mass                  2      40.000000
mass                  3      35.500000
mass                  4      15.999430   #O
mass                  5      28.085500   #Si

#############
pair_style            hybrid/overlay lj/cut/coul/long 9.0  deepmd myc.pb
Summary of lammps deepmd module ...
  >>> Info of deepmd-kit:
  installed to:       /home/ps/anaconda/envs/deepmd
  source:             v2.0.3-1-g96e623d5
  source branch:       HEAD
  source commit:      96e623d5
  source commit at:   2021-11-09 17:54:01 -0500
  surpport model ver.:1.0 
  build float prec:   double
  build with tf inc:  /home/ps/anaconda/envs/deepmd/include;/home/ps/anaconda/envs/deepmd/include
  build with tf lib:  /home/ps/anaconda/envs/deepmd/lib/libtensorflow_cc.so;/home/ps/anaconda/envs/deepmd/lib/libtensorflow_framework.so
  set tf intra_op_parallelism_threads: 0
  set tf inter_op_parallelism_threads: 0
  >>> Info of lammps module:
  use deepmd-kit at:  /home/ps/anaconda/envs/deepmd
  source:             v2.0.3-1-g96e623d5
  source branch:      HEAD
  source commit:      96e623d5
  source commit at:   2021-11-09 17:54:01 -0500
  build float prec:   double
  build with tf inc:  /home/ps/anaconda/envs/deepmd/include;/home/ps/anaconda/envs/deepmd/include
  build with tf lib:  /home/ps/anaconda/envs/deepmd/lib/libtensorflow_cc.so;/home/ps/anaconda/envs/deepmd/lib/libtensorflow_framework.so
2023-03-15 16:22:00.377638: I tensorflow/core/platform/cpu_feature_guard.cc:151] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  SSE4.1 SSE4.2 AVX AVX2 AVX512F FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2023-03-15 16:22:00.395970: I tensorflow/core/common_runtime/process_util.cc:146] Creating new thread pool with default inter op setting: 2. Tune using inter_op_parallelism_threads for best performance.
  >>> Info of model(s):
  using   1 model(s): myc.pb 
  rcut in model:      9
  ntypes in model:    3

pair_coeff             * *    deepmd myc.pb #na-na
WARNING: Interaction between types 0 and 4 is set with deepmd, but will be ignored.
 Deepmd model has only 3 types, it only computes the mulitbody interaction of types: 1-3. (/home/conda/feedstock_root/build_artifacts/libdeepmd_1638098235911/work/source/lmp/pair_deepmd.cpp:984)
WARNING: Interaction between types 0 and 5 is set with deepmd, but will be ignored.
 Deepmd model has only 3 types, it only computes the mulitbody interaction of types: 1-3. (/home/conda/feedstock_root/build_artifacts/libdeepmd_1638098235911/work/source/lmp/pair_deepmd.cpp:984)
WARNING: Interaction between types 1 and 4 is set with deepmd, but will be ignored.
 Deepmd model has only 3 types, it only computes the mulitbody interaction of types: 1-3. (/home/conda/feedstock_root/build_artifacts/libdeepmd_1638098235911/work/source/lmp/pair_deepmd.cpp:984)
WARNING: Interaction between types 1 and 5 is set with deepmd, but will be ignored.
 Deepmd model has only 3 types, it only computes the mulitbody interaction of types: 1-3. (/home/conda/feedstock_root/build_artifacts/libdeepmd_1638098235911/work/source/lmp/pair_deepmd.cpp:984)
WARNING: Interaction between types 2 and 4 is set with deepmd, but will be ignored.
 Deepmd model has only 3 types, it only computes the mulitbody interaction of types: 1-3. (/home/conda/feedstock_root/build_artifacts/libdeepmd_1638098235911/work/source/lmp/pair_deepmd.cpp:984)
WARNING: Interaction between types 2 and 5 is set with deepmd, but will be ignored.
 Deepmd model has only 3 types, it only computes the mulitbody interaction of types: 1-3. (/home/conda/feedstock_root/build_artifacts/libdeepmd_1638098235911/work/source/lmp/pair_deepmd.cpp:984)
WARNING: Interaction between types 3 and 4 is set with deepmd, but will be ignored.
 Deepmd model has only 3 types, it only computes the mulitbody interaction of types: 1-3. (/home/conda/feedstock_root/build_artifacts/libdeepmd_1638098235911/work/source/lmp/pair_deepmd.cpp:984)
WARNING: Interaction between types 3 and 5 is set with deepmd, but will be ignored.
 Deepmd model has only 3 types, it only computes the mulitbody interaction of types: 1-3. (/home/conda/feedstock_root/build_artifacts/libdeepmd_1638098235911/work/source/lmp/pair_deepmd.cpp:984)
WARNING: Interaction between types 4 and 4 is set with deepmd, but will be ignored.
 Deepmd model has only 3 types, it only computes the mulitbody interaction of types: 1-3. (/home/conda/feedstock_root/build_artifacts/libdeepmd_1638098235911/work/source/lmp/pair_deepmd.cpp:984)
WARNING: Interaction between types 4 and 5 is set with deepmd, but will be ignored.
 Deepmd model has only 3 types, it only computes the mulitbody interaction of types: 1-3. (/home/conda/feedstock_root/build_artifacts/libdeepmd_1638098235911/work/source/lmp/pair_deepmd.cpp:984)
WARNING: Interaction between types 5 and 5 is set with deepmd, but will be ignored.
 Deepmd model has only 3 types, it only computes the mulitbody interaction of types: 1-3. (/home/conda/feedstock_root/build_artifacts/libdeepmd_1638098235911/work/source/lmp/pair_deepmd.cpp:984)

pair_coeff            1 4    lj/cut/coul/long  0.01527718    2.4875 #o-na
pair_coeff            2 4    lj/cut/coul/long  0.01474814    2.7628 #o-ca
pair_coeff            3 4    lj/cut/coul/long  0.01453565    3.48 #o-cl

pair_coeff            1 5    lj/cut/coul/long  0.006399  3.084 #si-na
pair_coeff            2 5    lj/cut/coul/long  0.00617505    3.3593 #si-ca
pair_coeff            3 5    lj/cut/coul/long  0.00608832    4.0765 #si-cl

pair_coeff            4 4    lj/cut/coul/long  0.009891  2.86 #o-o
pair_coeff            4 5    lj/cut/coul/long  0.004143  3.4565 #o-si
pair_coeff            5 5    lj/cut/coul/long  0.001735  4.053 #si-si

kspace_style   pppm	1.0e-5

neighbor       2.0 bin
neigh_modify   delay 0 every 1 check yes

#----------Region--------------------
group                fixed  region   fixed
8 atoms in group fixed
group                SiO2  type     3 4
4819 atoms in group SiO2
group                Na    type        1
3647 atoms in group Na
group                Ca    type        2
1925 atoms in group Ca
group                Cl    type        3
4559 atoms in group Cl
group                L     type      1 2 3
10131 atoms in group L
group                shell   subtract  SiO2  fixed
4814 atoms in group shell

##----------------initialize-----------------------
thermo         1000
thermo_style   custom step temp density pe ke etotal press vol

velocity     all create 1200 23452342 rot yes dist gaussian mom yes
minimize        1.0e-4  1.0e-5   1000   10000

CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE

Your simulation uses code contributions which should be cited:
- USER-DEEPMD package:
The log file lists these citations in BibTeX format.

CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE-CITE

PPPM initialization ...
WARNING: System is not charge neutral, net charge = 2519.5932 (src/kspace.cpp:325)
  using 12-bit tables for long-range coulomb (src/kspace.cpp:340)
  G vector (1/distance) = 0.34163479
  grid = 75 75 75
  stencil order = 5
  estimated absolute RMS force accuracy = 0.00018814002
  estimated relative force accuracy = 1.3065601e-05
  using double precision FFTW3
  3d grid and FFT values/proc = 551368 421875
WARNING: Interaction between types 1 and 4 is set with deepmd, but will be ignored.
 Deepmd model has only 3 types, it only computes the mulitbody interaction of types: 1-3. (/home/conda/feedstock_root/build_artifacts/libdeepmd_1638098235911/work/source/lmp/pair_deepmd.cpp:1015)
WARNING: Interaction between types 1 and 5 is set with deepmd, but will be ignored.
 Deepmd model has only 3 types, it only computes the mulitbody interaction of types: 1-3. (/home/conda/feedstock_root/build_artifacts/libdeepmd_1638098235911/work/source/lmp/pair_deepmd.cpp:1015)
WARNING: Interaction between types 2 and 4 is set with deepmd, but will be ignored.
 Deepmd model has only 3 types, it only computes the mulitbody interaction of types: 1-3. (/home/conda/feedstock_root/build_artifacts/libdeepmd_1638098235911/work/source/lmp/pair_deepmd.cpp:1015)
WARNING: Interaction between types 2 and 5 is set with deepmd, but will be ignored.
 Deepmd model has only 3 types, it only computes the mulitbody interaction of types: 1-3. (/home/conda/feedstock_root/build_artifacts/libdeepmd_1638098235911/work/source/lmp/pair_deepmd.cpp:1015)
WARNING: Interaction between types 3 and 4 is set with deepmd, but will be ignored.
 Deepmd model has only 3 types, it only computes the mulitbody interaction of types: 1-3. (/home/conda/feedstock_root/build_artifacts/libdeepmd_1638098235911/work/source/lmp/pair_deepmd.cpp:1015)
WARNING: Interaction between types 3 and 5 is set with deepmd, but will be ignored.
 Deepmd model has only 3 types, it only computes the mulitbody interaction of types: 1-3. (/home/conda/feedstock_root/build_artifacts/libdeepmd_1638098235911/work/source/lmp/pair_deepmd.cpp:1015)
WARNING: Interaction between types 4 and 4 is set with deepmd, but will be ignored.
 Deepmd model has only 3 types, it only computes the mulitbody interaction of types: 1-3. (/home/conda/feedstock_root/build_artifacts/libdeepmd_1638098235911/work/source/lmp/pair_deepmd.cpp:1015)
WARNING: Interaction between types 4 and 5 is set with deepmd, but will be ignored.
 Deepmd model has only 3 types, it only computes the mulitbody interaction of types: 1-3. (/home/conda/feedstock_root/build_artifacts/libdeepmd_1638098235911/work/source/lmp/pair_deepmd.cpp:1015)
WARNING: Interaction between types 5 and 5 is set with deepmd, but will be ignored.
 Deepmd model has only 3 types, it only computes the mulitbody interaction of types: 1-3. (/home/conda/feedstock_root/build_artifacts/libdeepmd_1638098235911/work/source/lmp/pair_deepmd.cpp:1015)
WARNING: Impropers are defined but no improper style is set (src/force.cpp:206)
Unused restart file global fix info:
  fix style: nvt, fix ID: 1
  fix style: npt, fix ID: 2
Neighbor list info ...
  update every 1 steps, delay 0 steps, check yes
  max neighbors/atom: 2000, page size: 100000
  master list distance cutoff = 11
  ghost atom cutoff = 11
  binsize = 5.5, bins = 14 14 14
  3 neighbor lists, perpetual/occasional/extra = 3 0 0
  (1) pair lj/cut/coul/long, perpetual, skip from (3)
      attributes: half, newton on
      pair build: skip
      stencil: none
      bin: none
  (2) pair deepmd, perpetual
      attributes: full, newton on
      pair build: full/bin
      stencil: full/bin/3d
      bin: standard
  (3) neighbor class addition, perpetual, half/full from (2)
      attributes: half, newton on
      pair build: halffull/newton
      stencil: none
      bin: none
Setting up cg style minimization ...
  Unit style    : metal
  Current step  : 100000
Segmentation fault (core dumped)

**Comment by wanghan-iapcm** [comment]
The deepmd correctly ignores the interaction between type 0-2 with 3-5, you may see the warning 
```
WARNING: Interaction between types 0 and 4 is set with deepmd, but will be ignored.
Deepmd model has only 3 types, it only computes the mulitbody interaction of types: 1-3. (/home/conda/feedstock_root/build_artifacts/libdeepmd_1638098235911/work/source/lmp/pair_deepmd.cpp:984)
```

From the provided information I do not see why the seg fault happens.

**Comment by njzjz** [comment]
A bug was fixed in #1727. It looks like your version is too old.

**Reply by Nana-bai** [reply]
However, I can 't use conda to upgrade deepkit to 2.1.2 or higher, the display error is glibc version 2.31, please ask glibc version downgrade will affect other programs, glibc version must be 2.17 version ?
![1](https://user-images.githubusercontent.com/127933368/225621634-f60dc536-72f1-43ba-baf0-40a856a033ce.png)
![2](https://user-images.githubusercontent.com/127933368/225621635-bdf4ce48-8c17-4b1c-9801-25d926d14715.png)

**Reply by njzjz** [reply]
`lammps-dp` was renamed to `lammps` in recent versions.

**Reply by Nana-bai** [reply]
Ask, I upgraded to deepkit2.2.1, is deepmd unable to use hybrid deepmd lj command ?


#############
pair_style            hybrid deepmd 5yc.pb  lj/cut/coul/long 9.0
DeePMD-kit WARNING: Environmental variable TF_INTRA_OP_PARALLELISM_THREADS is not set. Tune TF_INTRA_OP_PARALLELISM_THREADS for the best performance. See https://deepmd.rtfd.io/parallelism/ for more information.
DeePMD-kit WARNING: Environmental variable TF_INTER_OP_PARALLELISM_THREADS is not set. Tune TF_INTER_OP_PARALLELISM_THREADS for the best performance. See https://deepmd.rtfd.io/parallelism/ for more information.
DeePMD-kit WARNING: Environmental variable OMP_NUM_THREADS is not set. Tune OMP_NUM_THREADS for the best performance. See https://deepmd.rtfd.io/parallelism/ for more information.
Summary of lammps deepmd module ...
  >>> Info of deepmd-kit:
  installed to:       /home/ps/anaconda/envs/deepmd
  source:             v2.2.1
  source branch:       HEAD
  source commit:      3ac8c4c7
  source commit at:   2023-03-16 12:33:24 +0800
  surpport model ver.:1.1 
  build variant:      cpu
  build with tf inc:  /home/ps/anaconda/envs/deepmd/include;/home/ps/anaconda/envs/deepmd/include
  build with tf lib:  /home/ps/anaconda/envs/deepmd/lib/libtensorflow_cc.so
  set tf intra_op_parallelism_threads: 0
  set tf inter_op_parallelism_threads: 0
  >>> Info of lammps module:
  use deepmd-kit at:  /home/ps/anaconda/envs/deepmdDeePMD-kit WARNING: Environmental variable TF_INTRA_OP_PARALLELISM_THREADS is not set. Tune TF_INTRA_OP_PARALLELISM_THREADS for the best performance. See https://deepmd.rtfd.io/parallelism/ for more information.
DeePMD-kit WARNING: Environmental variable TF_INTER_OP_PARALLELISM_THREADS is not set. Tune TF_INTER_OP_PARALLELISM_THREADS for the best performance. See https://deepmd.rtfd.io/parallelism/ for more information.
DeePMD-kit WARNING: Environmental variable OMP_NUM_THREADS is not set. Tune OMP_NUM_THREADS for the best performance. See https://deepmd.rtfd.io/parallelism/ for more information.
2023-03-17 15:55:42.117379: I tensorflow/core/platform/cpu_feature_guard.cc:193] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  SSE4.1 SSE4.2 AVX AVX2 AVX512F AVX512_VNNI FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2023-03-17 15:55:42.139464: I tensorflow/core/common_runtime/process_util.cc:146] Creating new thread pool with default inter op setting: 2. Tune using inter_op_parallelism_threads for best performance.
2023-03-17 15:55:43.737495: I tensorflow/compiler/mlir/mlir_graph_optimization_pass.cc:354] MLIR V1 optimization pass is not enabled
  >>> Info of model(s):
  using   1 model(s): 5yc.pb 
  rcut in model:      9
  ntypes in model:    5

pair_coeff            * *  #na-na
ERROR: Incorrect args for pair coefficients (src/pair_hybrid.cpp:472)
Last command: pair_coeff            * *  #na-na

**Reply by njzjz** [reply]
This error is reported by LAMMPS itself. It looks that the correct command is `pair_coeff * * deepmd`. You might first read LAMMPS documentation carefully.
